Adding dbt Metrics Layer page

runleonarun opened the pull request:
## Description & motivation
Adding a place for new metrics page and rethinking the way we present information.

## To-do before merge
<!---
(Optional -- remove this section if not needed)
Include any notes about things that need to happen before this PR is merged, e.g.:
- [ ] Change the base branch
- [ ] Ensure PR #56 is merged
-->

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [ ] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [x] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

## Checklist
If you added new pages (delete if not applicable):
- [ ] The page has been added to `website/sidebars.js`
- [ ] The new page has a unique filename

If you removed existing pages (delete if not applicable):
- [ ] The page has been removed from `website/sidebars.js`
- [ ] An entry has been added to `_redirects`

Commit:
trying to add new page for metrics

runleonarun commented on website/sidebars.js:
```diff
@@ -117,8 +117,15 @@ module.exports = {
+    {
+      type: "category",
+      label: "Introduction",
+      link: { type: 'doc', id: 'docs/dbt-cloud/cloud-intro' },
```
@JKarlavige Is it possible to upgrade our docusaurus to 2.0.0-beta.14? I want to use this link ability but can't currently use it (the build fails). See the [category doc link ](https://docusaurus.io/docs/sidebar#category-doc-link) docs for more context.

Thank you 🙏🏻

JKarlavige replied:
@runleonarun We are starting on the version update today and will be ready at the latest tomorrow morning!

Commit:
addign some metrics layer thoughts

drewbanin commented on website/docs/docs/dbt-cloud/cloud-metrics-layer.md:
```diff
@@ -0,0 +1,13 @@
+  - dbt metrics layer
+---
+
+[dbt metrics](/docs/building-a-dbt-project/metrics) keeps your metrics consistent and provides a way to standardize them for an organization by abstracting metrics calculations out of the data warehouse and into the transformation later, which is where dbt exists. dbt metrics solves the problem of trying to calculate metrics along a set of specified dimensions accurately when metrics values are c ...
```
this is a super-nit, but i think "metrics" here should be plural (like, the metrics themselves). "dbt metrics" is not going to be a singularized product name, we'll use that name to refer to the nodes in the dbt DAG themselves.

> by abstracting metrics calculations out of the data warehouse and into the transformation later, which is where dbt exists.

I really like this kind of framing! I think that "data warehouse" in this sentence kind of confuses the point though. Maybe we could position metrics in opposition to 1) pre-aggregation and 2) reimplementation in downstream data tools? So, something like:

> dbt metrics ensure metric consistency and provide a way to standardize metrics under version control in dbt projects. By abstracting metrics calculations out of pre-aggregated tables or specific BI tools, dbt metrics can be defined once and used everywhere. This helps ensure consistent reporting of key business metrics especially in an environment where metric definitions and dimensions are changing along with your business.

^ Feel free to workshop that - just a suggestion :)

runleonarun replied:
Addressed here: https://github.com/dbt-labs/docs.getdbt.com/pull/998/commits/79b74e0ef57dd36709435e4695677da326c9bfd2

drewbanin commented on website/docs/docs/dbt-cloud/cloud-metrics-layer.md:
```diff
@@ -0,0 +1,13 @@
+
+[dbt metrics](/docs/building-a-dbt-project/metrics) keeps your metrics consistent and provides a way to standardize them for an organization by abstracting metrics calculations out of the data warehouse and into the transformation later, which is where dbt exists. dbt metrics solves the problem of trying to calculate metrics along a set of specified dimensions accurately when metrics values are c ...
+
+Because dbt already transforms your data, it can use the same plane of existence to abstract out the metrics. dbt already knows about your core entities, for example, users, orders, and sessions. It can tell you not just about sessions, but average time on site. Or it might tell you not only about users but new users across these dimensions and this time range.
```
"plane of existence" is an amazing line haha

runleonarun replied:
You might not want to encourage me too much. Our docs will end up with the most [purple prose](https://en.wikipedia.org/wiki/Purple_prose) ever :)

drewbanin replied:
purple prose for purple people if you ask me

drewbanin commented on website/docs/docs/dbt-cloud/cloud-metrics-layer.md:
```diff
@@ -0,0 +1,13 @@
+
+Because dbt already transforms your data, it can use the same plane of existence to abstract out the metrics. dbt already knows about your core entities, for example, users, orders, and sessions. It can tell you not just about sessions, but average time on site. Or it might tell you not only about users but new users across these dimensions and this time range.
+
+[Sign up for the beta!](https://www.getdbt.com/signup/)
```
Could you link to https://forms.gle/4hi8YQ4mQ35QvYCh7 here? The CTA is to sign up for the beta, which isn't GA in dbt Cloud yet!

runleonarun replied:
Done! https://github.com/dbt-labs/docs.getdbt.com/pull/998/commits/392bd484e7a610b57e04ce7ab06690d9a72ad978

drewbanin reviewed (commented):
Couple of comments, but overall feeling really good! Thanks for opening this PR :)

runleonarun commented on website/docs/docs/dbt-cloud/cloud-metrics-layer.md:
```diff
@@ -0,0 +1,13 @@
+  - dbt metrics layer
+---
+
+[dbt metrics](/docs/building-a-dbt-project/metrics) keeps your metrics consistent and provides a way to standardize them for an organization by abstracting metrics calculations out of the data warehouse and into the transformation later, which is where dbt exists. dbt metrics solves the problem of trying to calculate metrics along a set of specified dimensions accurately when metrics values are c ...
```
```suggestion
[dbt metrics](/docs/building-a-dbt-project/metrics) ensure metric consistency and provide a way to standardize metrics under version control in dbt projects. By abstracting metrics calculations out of pre-aggregated tables or specific business intelligence tools (BI tools), dbt metrics can be defined once and used everywhere. This helps ensure consistent reporting of key business metrics especially in an environment where metric definitions and dimensions are changing along with your business.
```

runleonarun commented on website/docs/docs/dbt-cloud/cloud-metrics-layer.md:
```diff
@@ -0,0 +1,13 @@
+
+[dbt metrics](/docs/building-a-dbt-project/metrics) keeps your metrics consistent and provides a way to standardize them for an organization by abstracting metrics calculations out of the data warehouse and into the transformation later, which is where dbt exists. dbt metrics solves the problem of trying to calculate metrics along a set of specified dimensions accurately when metrics values are c ...
+
+Because dbt already transforms your data, it can use the same plane of existence to abstract out the metrics. dbt already knows about your core entities, for example, users, orders, and sessions. It can tell you not just about sessions, but average time on site. Or it might tell you not only about users but new users across these dimensions and this time range.
```
```suggestion
dbt already transforms your data, so it can use the same plane of existence to abstract out the metrics. Because dbt already knows about your core entities, such as users, orders, and sessions, dbt metrics can reveal a broader story. You can learn not just about sessions, but about average time on site, or not only about users but about new users across these dimensions and this time range.
```

runleonarun commented on website/docs/docs/dbt-cloud/cloud-metrics-layer.md:
```diff
@@ -0,0 +1,13 @@
+
+Because dbt already transforms your data, it can use the same plane of existence to abstract out the metrics. dbt already knows about your core entities, for example, users, orders, and sessions. It can tell you not just about sessions, but average time on site. Or it might tell you not only about users but new users across these dimensions and this time range.
+
+[Sign up for the beta!](https://www.getdbt.com/signup/)
```
```suggestion
[Sign up for the beta!](https://forms.gle/4hi8YQ4mQ35QvYCh7)
```

runleonarun committed:
Update website/docs/docs/dbt-cloud/cloud-metrics-layer.md

runleonarun committed:
Apply suggestions from code review

runleonarun commented on website/docs/docs/dbt-cloud/cloud-metrics-layer.md:
```diff
@@ -0,0 +1,13 @@
+
+[dbt metrics](/docs/building-a-dbt-project/metrics) ensure metric consistency and provide a way to standardize metrics under version control in dbt projects. By abstracting metrics calculations out of pre-aggregated tables or specific business intelligence tools (BI tools), dbt metrics can be defined once and used everywhere. This helps ensure consistent reporting of key business metrics especial ...
+
+dbt already transforms your data, so it can use the same plane of existence to abstract out the metrics. Because dbt already knows about your core entities, such as users, orders, and sessions, dbt metrics can reveal a broader story. You can learn not just about sessions, but about average time on site, or not only about users but about new users across these dimensions and this time range.
```
```suggestion
dbt already transforms your data, so it can use the same plane of existence to abstract out the metrics. Because dbt already knows about your core entities, such as users, orders, and sessions, dbt metrics can reveal a broader story. You can learn not just about sessions, but about average time on site, or not only about users but about new users across xyz dimensions and in x time range.
```

runleonarun committed:
Update website/docs/docs/dbt-cloud/cloud-metrics-layer.md

runleonarun commented on website/docs/docs/dbt-cloud/cloud-metrics-layer.md:
```diff
@@ -0,0 +1,13 @@
+  - dbt metrics layer
+---
+
+[dbt metrics](/docs/building-a-dbt-project/metrics) ensure metric consistency and provide a way to standardize metrics under version control in dbt projects. By abstracting metrics calculations out of pre-aggregated tables or specific business intelligence tools (BI tools), dbt metrics can be defined once and used everywhere. This helps ensure consistent reporting of key business metrics especial ...
```
```suggestion
[dbt metrics](/docs/building-a-dbt-project/metrics) ensure metric consistency and provide a way to standardize metrics under version control in dbt projects. By abstracting metrics calculations out of pre-aggregated tables or specific business intelligence tools (BI tools), dbt metrics can be defined once and used everywhere. Defining metrics in one place ensures consistent reporting of key business metrics especially in an environment where metric definitions and dimensions are changing along with your business.
```

runleonarun committed:
Update website/docs/docs/dbt-cloud/cloud-metrics-layer.md

runleonarun commented:
@drewbanin I added your suggestions and edited a few more things. Is this all we want here? It feels like we could unpack a few things:

* I'm curious about a specific example for this line, "You can learn not just about sessions, but about average time on site, or not only about users but about new users across xyz dimensions and in x time range." 

* Also, why should I be excited about the metrics layer? For example, when I read 
  > Defining metrics in one place ensures consistent reporting of key business metrics especially in an environment where metric definitions and dimensions are changing along with your business.

  it sounds important but I wonder if a concrete example would help here too.  
 
* Do you have a demo we can link to per @krevitt's request?
I think that's it for now 😅

Commit:
Merge branch 'current' into adding-metrics-layer-page

Commit:
getting more examples with Drew

runleonarun commented on website/sidebars.js:
```diff
@@ -117,8 +117,16 @@ module.exports = {
+    {
+      type: "category",
+      label: "Overview",
+//      link: { type: 'doc', id: 'docs/dbt-cloud/cloud-overview' }, 
```
```suggestion
     link: { type: 'doc', id: 'docs/dbt-cloud/cloud-overview' }, 
```

runleonarun committed:
Update website/sidebars.js

runleonarun commented on website/sidebars.js:
```diff
@@ -117,8 +117,16 @@ module.exports = {
+    {
+      type: "category",
+      label: "Overview",
+     link: { type: 'doc', id: 'docs/dbt-cloud/cloud-overview' }, 
```
```suggestion
      link: { type: 'doc', id: 'docs/dbt-cloud/cloud-overview' }, 
```

runleonarun committed:
Update website/sidebars.js

Commit:
Merge branch 'current' into adding-metrics-layer-page

Commit:
Merge branch 'adding-metrics-layer-page' of github.com:dbt-labs/docs.getdbt.com into adding-metrics-layer-page

runleonarun commented on website/sidebars.js:
```diff
@@ -117,8 +117,16 @@ module.exports = {
+      label: "Overview",
+      link: { type: 'doc', id: 'docs/dbt-cloud/cloud-overview' }, 
+      items: [
+        'docs/dbt-cloud/cloud-overview', 
```
```suggestion
```

runleonarun committed:
Update website/sidebars.js

We no longer need this as a separate page because you see it when you click "Overview"

runleonarun commented on website/docs/docs/dbt-cloud/cloud-overview.md:
```diff
@@ -1,40 +1,41 @@
 
 dbt Cloud’s generous Developer (free) plan and deep integration with dbt Core make it well suited for data teams small and large alike.  
 
 [Get started with dbt Cloud](https://www.getdbt.com/signup/)
```
```suggestion
You can get started with dbt Cloud by [Signing up](https://www.getdbt.com/signup/).
```

runleonarun committed:
Update website/docs/docs/dbt-cloud/cloud-overview.md

Commit:
moving metrics layer

Commit:
adding a landign page for cloud. scope creep

drewbanin commented on website/sidebars.js:
```diff
@@ -146,7 +152,14 @@ module.exports = {
       type: "category",
       label: "Using dbt Cloud",
+        link: {
+        type: 'generated-index',
```
whoaaah - is this new?? Love it - just noting that the styles are borked on this page (though you probably already know that 😄 ) https://deploy-preview-998--docs-getdbt-com.netlify.app/docs/dbt-cloud

runleonarun replied:
@drewbanin Yes it's new! Just trying to fix those fonts/colors now.

Commit:
removing new landing page for now

drewbanin reviewed (approved):
ship it!
